crates/api_models/src/admin.rs

```diff
@@ -2592,6 +2592,44 @@ pub struct BusinessPayoutLinkConfig {
     pub payout_test_mode: Option<bool>,
 }
 
+#[derive(Clone, Debug, serde::Deserialize, serde::Serialize, ToSchema)]
+pub struct OutgoingWebhookCustomHttpHeaders {
+    #[serde(flatten)]
+    pub headers: HashMap<String, Secret<String>>,
+}
+
+impl OutgoingWebhookCustomHttpHeaders {
+    fn mask_value(&self, key: &str) -> Secret<String> {
+        let key_len = key.len();
+        let masked_string = if key_len <= 4 {
+            "*".repeat(key_len)
+        } else {
+            // Show the first two and last two characters, mask the rest with '*'
+            let mut masked_string = String::new();
+            // Iterate through characters by their index
+            for (index, character) in key.chars().enumerate() {
+                if index < 2 || index >= key_len - 2 {
+                    masked_string.push(character); // Keep the first two and last two characters
+                } else {
+                    masked_string.push('*'); // Mask the middle characters
+                }
+            }
+            masked_string
+        };
+        Secret::new(masked_string)
+    }
+    pub fn get_masked_headers(&self) -> Self {
+        let mut masked_headers_hash_map: HashMap<String, Secret<String>> = HashMap::new();
+        for header in &self.headers {
+            let masked_value = self.mask_value(header.1.peek());
+            masked_headers_hash_map.insert(header.0.clone(), masked_value);
+        }
+        Self {
+            headers: masked_headers_hash_map,
+        }
+    }
```

Comment:
Ummm, this allows the possibility of `get_masked_headers()` being called repetitively on an instance of `OutgoingWebhookCustomHttpHeaders` with no additional effect? 
Let's avoid this being called on `self`, it should be an associated function.

Also, this would also allow us to construct `OutgoingWebhookCustomHttpHeaders` without having called `get_masked_headers()` and we'd end up serializing plaintext values instead.

Something like:

```rust
#[derive(Debug, serde::Serialize)]
pub struct MaskedHeaders {
    #[serde(flatten)]
    headers: HashMap<String, Secret<String>>,
}

impl MaskedHeaders {
    fn mask_value(value: &str) -> Secret<String> {
        let value_len = value.len();

        let masked_value = if value_len <= 4 {
            "*".repeat(value_len)
        } else {
            value
                .char_indices()
                .into_iter()
                .map(|(index, ch)| {
                    if index < 2 || index >= value_len - 2 {
                        // Show the first two and last two characters, mask the rest with '*'
                        ch
                    } else {
                        // Mask the remaining characters
                        '*'
                    }
                })
                .collect::<String>()
        };

        Secret::new(masked_value)
    }

    pub fn from_headers(headers: HashMap<String, Secret<String>>) -> Self {
        let masked_headers = headers
            .into_iter()
            .map(|(key, value)| (key, Self::mask_value(value.peek())))
            .collect();

        Self {
            headers: masked_headers,
        }
    }
}
```

The above implementation has the added advantage that an instance of `MaskedHeaders` (constructed using `MaskedHeaders::from_headers()`) is guaranteed to always have masked values.